lib/src/runner/hybrid_listener.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:async";
+import "dart:isolate";
+
+import "package:async/async.dart";
+import "package:stack_trace/stack_trace.dart";
+import "package:stream_channel/stream_channel.dart";
+
+import "../util/remote_exception.dart";
+import "../utils.dart";
+
+/// A sink transformer that wraps data and error events so that errors can be
+/// decoded after being JSON-serialized.
+final _transformer = new StreamSinkTransformer.fromHandlers(
+    handleData: (data, sink) {
+      ensureJsonEncodable(data);
```

Comment:
This is called synchronously from the user's `Sink.add()` call, so if `ensureJsonEncodable()` throws here they'll get a helpful stack trace. I'll add a comment to that effect.